src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -442,23 +441,7 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
             .state_store()
             .clone();
         let table_id = self.backfill_state_store.state_store().table_id().into();
-        static STATE_TABLE_INITIALIZED: Once = Once::new();
-        tokio::spawn(async move {
-            // This is for self.backfill_finished() to be safe.
-            // We wait for 1st epoch's curr, i.e., the 2nd epoch's prev.
-            let epoch = first_epoch.curr;
-            tracing::info!("waiting for epoch: {}", epoch);
-            state_store
-                .try_wait_epoch(
-                    HummockReadEpoch::Committed(epoch),
-                    TryWaitEpochOptions { table_id },
-                )
-                .await
-                .expect("failed to wait epoch");
-            STATE_TABLE_INITIALIZED.call_once(|| ());
```

Comment:
Was it correct to use the global static variable `STATE_TABLE_INITIALIZED` previously? 

IIUC, Since it exists globally, it can be called at most once in the whole lifetime of a CN process, even if we have multiple source created, and then it will always be immediately completed for subsequent sources after the first source gets completed.

If it's incorrect, we may need to fix it in previous release.

Reply:
🤡 You are right.

Reply:
But this is hard to trigger for serious workload though. Will only happen if the Kafka topic doesn't have new incoming data, and then scaled